ApollosProject/apollos-prototype
# Adds TextFeature resolver/schema/dataSource

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds the ability for ContentSeriesItems and WeekendContentItems to contain TextFeatures. if they have a custom rock field with a key of TextFeatures. 

![image](https://user-images.githubusercontent.com/1637694/60359275-9c745000-99a6-11e9-950d-9ffc22c43e97.png)

### What design trade-offs/decisions were made?

- Front end work is covered in a different issue / pr. 
- Punted the implementation of the "order" field. 


### How do I test this PR?

1. Run the following query in GQL
```
{
  userFeed {
    edges {
      node {
        title
        __typename
        ... on ContentSeriesContentItem {
          features {
            id
            ... on TextFeature {
              id
              body
            }
          }
        }
        
      }
    }
  }
}
```

You will see a text feature listed under the content item titled "The New Weekend Experience"

### What automated tests did you write?

Tests over all new dataSource functionality. 

## TODO:

- [x] Closes #802 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
